### PR TITLE
[Improvement-12029][dolphinscheduler-api] Upgrade the snakeyaml version to avoid vulnerabilities

### LINKED ISSUE #12029: [Dependencies] upgrade to snakeyaml 1.32 due to CVE
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

another snakeyaml release v1.32 to fix https://github.com/advisories/GHSA-9w3m-gqgf-c4p9

### What you expected to happen

n/a

### How to reproduce

n/a

### Anything else

_No response_

### Version

dev

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by seedscoder**
## Purpose of the pull request

close: #12029 

Upgrade the snakeyaml version to avoid vulnerabilities

## Brief change log

- Exclude existing indirect dependencies
- Use the new version of snakeyaml

### CODE REVIEW

**Review by caishunfeng — DISMISSED** [review]
LGTM if CI pass

**Comment by seedscoder** [comment]
@caishunfeng The version currently used by other components is vulnerable. Do I need to modify it? Are there any risks?
Snipaste_2022-11-05_09-41-31

In addition, how does CI retry?
Snipaste_2022-11-05_09-41-31

**Comment by kezhenxu94** [comment]
> @caishunfeng The version currently used by other components is vulnerable. Do I need to modify it? Are there any risks?

If you want, you can try to upgrade and test them.

File: dolphinscheduler-api/pom.xml [hunk 1]
```diff
@@ -142,6 +146,12 @@
         <dependency>
             <groupId>io.fabric8</groupId>
             <artifactId>kubernetes-client</artifactId>
+            <exclusions>
+                <exclusion>
+                    <groupId>org.yaml</groupId>
+                    <artifactId>snakeyaml</artifactId>
+                </exclusion>
+            </exclusions>
```
Comment (kezhenxu94) [inline dolphinscheduler-api/pom.xml]: You don't need this.

File: dolphinscheduler-api/pom.xml [hunk 2]
```diff
@@ -194,6 +204,11 @@
             <groupId>org.springdoc</groupId>
             <artifactId>springdoc-openapi-ui</artifactId>
         </dependency>
+
+        <dependency>
+            <groupId>org.yaml</groupId>
+            <artifactId>snakeyaml</artifactId>
+        </dependency>
```
Comment (kezhenxu94) [inline dolphinscheduler-api/pom.xml]: You don't need this

File: dolphinscheduler-api/pom.xml [hunk 3]
```diff
@@ -99,6 +99,10 @@
                     <groupId>org.apache.logging.log4j</groupId>
                     <artifactId>log4j-to-slf4j</artifactId>
                 </exclusion>
+                <exclusion>
+                    <groupId>org.yaml</groupId>
+                    <artifactId>snakeyaml</artifactId>
+                </exclusion>
```
Comment (kezhenxu94) [inline dolphinscheduler-api/pom.xml]: You don't need this

File: dolphinscheduler-bom/pom.xml [hunk 4]
```diff
@@ -94,7 +94,7 @@
         <snappy.version>1.1.8.4</snappy.version>
         <spark.version>3.2.2</spark.version>
         <janino.version>3.0.16</janino.version>
-        <snakeyaml.version>1.31</snakeyaml.version>
+        <snakeyaml.version>1.33</snakeyaml.version>
```
Comment (kezhenxu94) [inline dolphinscheduler-bom/pom.xml:97]: This is the only one place you need to modify,

**Review by kezhenxu94 — CHANGES_REQUESTED** [review]

**Comment by caishunfeng** [comment]
> @caishunfeng The version currently used by other components is vulnerable. Do I need to modify it? Are there any risks? 

The `OWASP` is not a requirement for the CI pass, it can be fixed if you are interested.

> In addition, how does CI retry? 

I had rerun the e2e CI.

**Review by kezhenxu94 — APPROVED** [review]

**Pull request merged**